tests/nominal/test_GroupRareLevelsTransformer.py

```diff
@@ -2,28 +2,26 @@
 import pandas as pd
 import pytest
 import test_aide as ta
+from test_BaseNominalTransformer import GenericBaseNominalTransformerTests
 
 import tests.test_data as d
-import tubular
+from tests.base_tests import (
+    ColumnStrListInitTests,
+    GenericFitTests,
+    GenericTransformTests,
+    OtherBaseBehaviourTests,
+    WeightColumnFitMixinTests,
+    WeightColumnInitMixinTests,
+)
 from tubular.nominal import GroupRareLevelsTransformer
 
 
-class TestInit:
+class TestInit(ColumnStrListInitTests, WeightColumnInitMixinTests):
```

Comment:
Q: do these test classes not need to inherit from the BaseNominalTransformer classes?

Reply:
test_BaseNominalTransformer only defines specific tests for the transform method.  I guess there might be a case for creating placeholder test classes for BaseNominalTransformerInitTests and BaseNominalTransformeFitTests in case some were added in future and child classes would be set to inherit by default?

Reply:
ah sorry I was looking at the e.g. 'GenericTransformTests' format classes in that file - but that raises a point, shouldn't these classes be inheriting from test_BaseNominalTransformer.GenericTransformTests/GenericInitTests/GenericFitTests rather than the base_tests versions? then as you say would future proof it and tests added in future would be auto-inherited

Reply:
Those classes aren't modified in the test_BaseNominalTransformer module.  We would need to inherit a class here which is created in that module and inherits from the base classes, hence the suggestion we create placeholder classes above.

Reply:
I have renamed GenericBaseNominalTransformerTests to BaseNominalTransformTests class for clarity [here](https://github.com/lvgig/tubular/pull/259/commits/70892b5d7d1b76be0971a4bf98b96b796e899ef1)

Reply:
Have also made GenericNominalTransformTests inherit from GenericTransformTests and removed multi inheritance in the child.  https://github.com/lvgig/tubular/pull/259/commits/913254eaee200c082f4f33a82c1b71cfd35a6ea1

Reply:
ahh sorry think I was muddled with classes, am with you now, looks good thanks!